cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -300,6 +350,215 @@ private void computeIncomingSummaryHelper(EndPointField destination, Operation c
+   * original read operation creates fields 'offset' and 'body' and request is for getting the
+   * outgoing operations for 'body', the 'offset' field will be eliminated from the output of read.
+   *
+   * @param sourceField the field belonging to the EndPoint for which outgoing operations to be returned
+   * @return outgoing operations
+   */
+  public Set<Operation> getOutgoingOperationsFromField(EndPointField sourceField) {
+    if (readOperations == null) {
+      // make sure non-transient fields are computed
+      computeAndValidateFieldLineageInfo(this.operations);
+    }
+
+    Map<String, Operation> result = new HashMap<>();
+    for (ReadOperation read : readOperations) {
+      if (read.getSource().equals(sourceField.getEndPoint()) && read.getOutputs().contains(sourceField.getField())) {
+        ReadOperation readToAdd = new ReadOperation(read.getName(), read.getDescription(), sourceField.getEndPoint(),
+                                                    sourceField.getField());
+        result.put(readToAdd.getName(), readToAdd);
+        Set<String> operationsToProcess = new HashSet<>(operationOutgoingConnections.get(read.getName()));
+        for (String operationToProcess : operationsToProcess) {
+          outgoingOperationsFromFieldHelper(operationToProcess, readToAdd, result);
+        }
+      }
+    }
+    return new HashSet<>(result.values());
+  }
+
+  private void outgoingOperationsFromFieldHelper(String currentOperationNameToProcess, Operation previousOperation,
+                                                 Map<String, Operation> result) {
+
+    Operation currentOperation = operationsMap.get(currentOperationNameToProcess);
+
+    List<String> commonFields = commonFieldsBetweenOperations(previousOperation, currentOperation);
+    if (commonFields.isEmpty()) {
+      return;
+    }
+
+    List<InputField> inputFields = new ArrayList<>();
+    for (String field : commonFields) {
+      inputFields.add(InputField.of(previousOperation.getName(), field));
+    }
+
+    Operation operationToAdd = null;
+
+    if (currentOperation.getType() == OperationType.TRANSFORM) {
+      TransformOperation transform = (TransformOperation) currentOperation;
+      operationToAdd = new TransformOperation(transform.getName(), transform.getDescription(), inputFields,
+                                              transform.getOutputs());
+    } else if (currentOperation.getType() == OperationType.WRITE) {
+      WriteOperation write = (WriteOperation) currentOperation;
+      operationToAdd = new WriteOperation(write.getName(), write.getDescription(), write.getDestination(),
+              inputFields);
+    }
+
+    if (operationToAdd != null) {
+      Operation operation = result.get(operationToAdd.getName());
+      if (operation != null) {
+        // operation was already added as a part of another path
+        // create new operation after merging the input and output fields of both
```

Comment:
I am not sure I understand this... why do we have to merge it with itself? If it has already been added, then what more is there to do?